python/source/org/libreoffice/ide/eclipse/python/PythonBuilder.java

```diff
@@ -0,0 +1,322 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+ * MA 02111-1307 USA
+ *
+ * The Initial Developer of the Original Code is: Sun Microsystems, Inc..
+ *
+ * Copyright: 2002 by Sun Microsystems, Inc.
+ *
+ * All Rights Reserved.
+ *
+ * Contributor(s): Cedric Bosdonnat
+ *
+ *
+ ************************************************************************/
+package org.libreoffice.ide.eclipse.python;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.LineNumberReader;
+import java.text.MessageFormat;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IFolder;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.core.runtime.Path;
+import org.eclipse.jdt.core.IClasspathEntry;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.JavaCore;
+import org.eclipse.jdt.core.JavaModelException;
+import org.eclipse.jdt.launching.IVMInstall;
+import org.eclipse.jdt.launching.JavaRuntime;
+import org.eclipse.jdt.ui.jarpackager.IJarExportRunnable;
+import org.eclipse.jdt.ui.jarpackager.JarPackageData;
+import org.libreoffice.ide.eclipse.core.PluginLogger;
+import org.libreoffice.ide.eclipse.core.model.IUnoidlProject;
+import org.libreoffice.ide.eclipse.core.model.ProjectsManager;
+import org.libreoffice.ide.eclipse.core.model.config.IOOo;
+import org.libreoffice.ide.eclipse.core.model.config.ISdk;
+import org.libreoffice.ide.eclipse.core.model.language.ILanguageBuilder;
+import org.libreoffice.ide.eclipse.core.model.utils.SystemHelper;
+import org.libreoffice.ide.eclipse.python.build.FilesVisitor;
+import org.libreoffice.ide.eclipse.python.build.UnoManifestProvider;
+import org.libreoffice.plugin.core.model.UnoPackage;
+
+/**
+ * The language builder implementation for Java.
+ */
+public class PythonBuilder implements ILanguageBuilder {
+
+    private Language mLanguage;
+
+    /**
+     * Constructor.
+     *
+     * @param pLanguage the Java Language object
+     */
+    public PythonBuilder(Language pLanguage) {
+        mLanguage = pLanguage;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public IFile createLibrary(IUnoidlProject pUnoProject) throws Exception {
+        IFile jarFile = ((PythonProjectHandler) mLanguage.getProjectHandler()).getJarFile(pUnoProject);
```

Comment:
So as mentioned to fill the oxt with Python files during the creation of the library, it is not possible because of the following

org.libreoffice.ide.eclipse.core.gui.PackageContentSelector
		UnoPackage pack = null;

        File prjFile = SystemHelper.getFile(pProject);

        // Export the library
        IFile library = null;
        ILanguageBuilder langBuilder = pProject.getLanguage().getLanguageBuidler();
        library = langBuilder.createLibrary(pProject);

        // Create the package model
        pack = UnoidlProjectHelper.createMinimalUnoPackage(pProject, pDestFile);

        if (library.exists()) {
            pack.addToClean(SystemHelper.getFile(library));
            File libraryFile = SystemHelper.getFile(library);
            pack.addFile(UnoPackage.getPathRelativeToBase(libraryFile, prjFile), libraryFile);
        }

As you can see the the oxt package is being created after the creation of the library, and also if we change the order of the calling statements, we have to change the signature, (to send the package pointer to the library creation function : createLibrary(pProject);) which we have to do in the Core which will result in changing in Java part so dropping the plan to include the Python in the create Library.

One other way is to include the Python files in this line only [pack = UnoidlProjectHelper.createMinimalUnoPackage(pProject, pDestFile);], in this there is a call to the Python module and we can make the necessary changes as we see fit.